test/unit/range/view/view_single_pass_input_test.cpp

```diff
@@ -103,21 +106,22 @@ TYPED_TEST(single_pass_input, view_concept)
 
 TYPED_TEST(single_pass_input, view_construction)
 {
-    using view_t = detail::single_pass_input_view<std::add_lvalue_reference_t<TypeParam>>;
+    using rng_t = decltype(std::declval<TypeParam>() | view::persist);
+    using view_t = detail::single_pass_input_view<rng_t>;
     EXPECT_TRUE(std::is_default_constructible_v<view_t>);
     EXPECT_TRUE(std::is_copy_constructible_v<view_t>);
     EXPECT_TRUE(std::is_move_constructible_v<view_t>);
     EXPECT_TRUE(std::is_copy_assignable_v<view_t>);
     EXPECT_TRUE(std::is_move_assignable_v<view_t>);
     EXPECT_TRUE(std::is_destructible_v<view_t>);
 
-    {  // from lvalue
+    {  // from lvalue container
         TypeParam p{this->data};
         [[maybe_unused]] detail::single_pass_input_view v{p};
```

Comment:
We should add a test to check if the deduced type is really the one we would like to see.
In this case check if `std::is_same_v<decltype(v), detail::single_pass_input_view<decltype(p | view::all)>>;` is true